src/Microsoft.AspNet.Server.Kestrel/Http/FrameHeaders.cs

```diff
@@ -152,7 +152,7 @@ IEnumerator IEnumerable.GetEnumerator()
             StringValues value;
             return
                 TryGetValueFast(item.Key, out value) &&
-                object.Equals(value, item.Value) &&
+                value.Equals(item.Value) &&
```

Comment:
The old check here could never return true. You're boxing two structs to compare them for reference equality. That's always false.